src/main/java/com/epam/lowcost/DAO/TicketDAOImpl.java

```diff
@@ -0,0 +1,217 @@
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        ticket = null;
+        return ticket;
+    }
+
+    @Override
+    public Ticket getById(long currentId) {
+        Ticket ticket = null;
+        String sql = String.format("SELECT * FROM TICKETS " +
+                "JOIN  USERS ON TICKETS.userId=USERS.id " +
+                "JOIN  FLIGHTS ON TICKETS.flightId=FLIGHTS.id " +
+                "JOIN  PLANES ON FLIGHTS.plane_id = PLANES.id " +
+                "WHERE TICKETS.isDeleted=false and TICKETS.id=%d", currentId);
+        try (Connection conn = dataSource.getConnection();
+             Statement stm = conn.createStatement();
+             ResultSet rs = stm.executeQuery(sql)) {
+            while (rs.next()) {
+                long id = rs.getLong("id");
+                Flight flight = extractFlightFromRS(rs);
+                User user = extractUserFromRS(rs);
+                boolean isBusiness = rs.getBoolean("isBusiness");
+                boolean hasLuggage = rs.getBoolean("hasLuggage");
+                boolean placePriority = rs.getBoolean("placePriority");
+                long price = rs.getLong("price");
+                ticket = new Ticket(id, user, flight, isBusiness, hasLuggage, placePriority, price, false);
+            }
+        }
+
+        catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return ticket;
+    }
+
+    @Override
+    public Ticket updateTicket(Ticket ticket) {
+
+        String sql = String.format("UPDATE TICKETS SET hasLuggage='%b',placePriority='%b'," +
+                        "isBusiness='%b', price='%d' WHERE id = %d",
+                ticket.isHasLuggage(),
+                ticket.isPlacePriority(),
+                ticket.isBusiness(),
+                ticket.getPrice(),
+                ticket.getId());
+
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                return ticket;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+
+        }
+        ticket = null;
+        return ticket;
+    }
+
+    @Override
+    public Ticket deleteTicket(long id) {
+        Ticket ticket = getById(id);
+        if (ticket == null)
+            return null;
+        ticket.setDeleted(true);
+        String sql = String.format("UPDATE TICKETS SET isDeleted = TRUE WHERE id = '%d'", id);
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                return ticket;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        ticket = null;
+        return ticket;
+    }
+
+
+    private User extractUserFromRS(ResultSet rs) throws SQLException {
```

Comment:
it is used in another class. Why you duplicate code?

Reply:
Remove all duplications of this code, not only in my class and merge with Nastya's updated branch.